core/src/main/java/com/amplifyframework/core/model/query/predicate/ContainsQueryOperator.java

```diff
@@ -40,6 +40,17 @@ public Object value() {
         return value;
     }
 
+    /**
+     * Returns true if the the provided field value contains
+     * the value associated with this operator.
+     * @param field the field value to operate on
+     * @return evaluated result of the operator
+     */
+    @Override
+    public boolean evaluate(String field) {
```

Comment:
Hm, this one I'm not as sold on.
Strings can contain other strings. But collections can contain items. So the choice of `String` here is somewhat arbitrary. Couldn't `ContainsQueryOperation` just easily have `implements QueryOperator<Collection<T>>` and ask `collection.contains( /* (T) */ fieldForEvalution)`?

Reply:
This is something I was discussing with others, and I'm not decided yet if I would rather create a separate operator for this purpose. It would be nice if there is an interface for `Container` to group String and Collection together hmm

Reply:
I think we'll probably have to have `StringContainsQueryOperator implements QueryOperator<String>`, and `CollectionContainsQueyrOperator implements QueryOperator<Collection<?>>`, as two separate things. Otherwise, the implementation will have a `String value`, and a `Collection<?> value`, and would need to conditionally use one or the other, which feels like an anti-pattern.

Reply:
For now, it will only operate on `String`, since current storage engine implementation doesn't support predicate on `Collection` yet

Reply:
Okay, sounds good